server/controllers/auth/signUp.ts

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable camelcase */
+import { NextFunction, Request, Response } from 'express';
+import { hash } from 'bcryptjs';
+import { sign } from 'jsonwebtoken';
+import { signUpSchema, CustomedError } from '../../utils';
+import { Donor } from '../../database/models';
+
+require('env2')('.env');
+
+const signUp = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const validationResult = await signUpSchema.validateAsync(req.body);
+    const {
+      name,
+      email,
+      password,
+      phone,
+      address,
+    } = validationResult;
+
+    const checkEmail = await Donor.findOne({
+      where: {
+        email,
+      },
+    });
+
+    if (checkEmail) {
+      throw new CustomedError(
+        'Email is used try another one',
+        400,
+      );
+    }
+    const hashedPassword = await hash(password, 10);
+
+    const donor: any = await Donor.create({
+      name, email, password: hashedPassword, phone, address,
+    });
+
+    // eslint-disable-next-line max-len
+    await sign({ id: donor.id }, process.env.SECRET as string, (error:Error | null, token: string | undefined) => {
```

Comment:
Here, I create function to sign token, so try to pull main in main and use it in your branch